module/VuFind/src/VuFind/Db/Migration/MigrationManager.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+/**
+ * Database migration manager.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Db
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Db\Migration;
+
+use Doctrine\DBAL\Connection;
```

Comment:
Should we use VuFind's connection here?
```suggestion
use VuFind\Db\Connection;
```

Reply:
I intentionally used the more generic class, since there is no dependency on VuFind-specific behavior here (since, at present, there is no VuFind-specific connection behavior).

Reply:
My reasoning here would be that this creates a direct dependency to Doctrine that could be avoided. But I have not problem with this if you prefer it that way.

Reply:
Since the code is using Doctrine interfaces for everything, I think it's already unavoidably dependent on the framework, and would have to be rewritten if we switched to a different framework in future. But you do make a point that using the VuFind namespace might prevent us from having to change the constructor signature in the future. Not sure that makes much difference, though, as this is not a class I would expect people to locally override very often. Guess I could still go either way on this!

Reply:
Even if we use a the Doctrine interfaces we could potentially keep them in the VuFind class and map them to a potential new solution. 

But I also don't think that anyone will customize this code. So, no strong feelings about this.

Reply:
I went ahead and made the change. It doesn't really make much difference, but as you say, this might save us trouble in the future, so why not? :-)